app/helpers/payments.ts

```diff
@@ -8,50 +8,96 @@ import {
   Operation,
   TransactionBuilder,
   BASE_FEE,
+  xdr,
 } from "stellar-sdk";
 import { IPaymentSummary } from "../interfaces/payments";
+import IWallet from "@component/interfaces/wallet";
 
 const server = new Server(process.env.NEXT_PUBLIC_SERVER_URL || "");
 
 const signTransaction = (transaction: Transaction, sourceKeys: Keypair) =>
   transaction.sign(sourceKeys);
 
-const submitTransaction = (transaction: Transaction) =>
-  server.submitTransaction(transaction);
+export const submitTransaction = async (transaction: Transaction) => {
+  try {
+    await server.submitTransaction(transaction);
+  } catch (error) {
+    console.error(error);
+  }
+};
 
 const loadAccount = async (publicKey: string) => server.loadAccount(publicKey);
 
 export const sendPayment: (paymentSummary: IPaymentSummary) => void = async ({
   signerKey,
   destinationPublicKey,
   amount,
-  memo
+  memo,
 }) => {
   const sourceKeys: Keypair = Keypair.fromSecret(signerKey);
-  const timeOutInSeconds: number = 60;
 
   try {
     await loadAccount(destinationPublicKey);
-    const sourceAccount = await loadAccount(sourceKeys.publicKey());
-
-    const transaction = new TransactionBuilder(sourceAccount, {
-      fee: BASE_FEE,
-      networkPassphrase: Networks.TESTNET,
-    })
-      .addOperation(
-        Operation.payment({
-          destination: destinationPublicKey,
-          asset: Asset.native(),
-          amount: amount,
-        })
-      )
-      .addMemo(Memo.text(memo))
-      .setTimeout(timeOutInSeconds)
-      .build();
+
+    const transaction = await buildTransaction(
+      sourceKeys.publicKey(),
+      destinationPublicKey,
+      amount,
+      memo
+    );
 
     signTransaction(transaction, sourceKeys);
     await submitTransaction(transaction);
   } catch (error) {
     console.error(error);
   }
 };
+
+export const buildTransaction = async (
+  sourcePublicKey: string,
+  destinationPublicKey: string,
+  amount: string,
+  memo: string
+) => {
+  const sourceAccount = await loadAccount(sourcePublicKey);
+
+  const transaction = new TransactionBuilder(sourceAccount, {
+    fee: BASE_FEE,
+    networkPassphrase: Networks.TESTNET,
+  })
+    .addOperation(
+      Operation.payment({
+        destination: destinationPublicKey,
+        asset: Asset.native(),
+        amount: amount,
+      })
+    )
+    .addMemo(Memo.text(memo))
```

Comment:
Memo should be optional

Reply:
I have changed "memo" to be an optional argument and added an empty string in case the memo does not exist.